components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/mail/DefaultEmailSendingModule.java

```diff
@@ -73,9 +73,20 @@ public void sendEmail() {
 
         try {
             if (this.notification == null) {
-                throw new NullPointerException("Notification not set. Please set the notification before sending messages");
+                throw new IllegalStateException("Notification not set. " +
+                        "Please set the notification before sending messages");
             }
             PrivilegedCarbonContext.startTenantFlow();
+            if (notificationData != null) {
+                String tenantDomain = notificationData.getDomainName();
+                PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+                carbonContext.setTenantDomain(tenantDomain);
+                carbonContext.getTenantId(true);
```

Comment:
When tenantDomain is already set to the carbonContext, getTenantId(boolean resolve) method in PrivilegedCarbonContext class resolves the tenantId and set it to the carbonContext. Here the get is called in order to set the tenantId to the carbonContext since we don't have the tenantId in hand and tenantDomain is set.